server/src/main/java/org/elasticsearch/common/breaker/ChildMemoryCircuitBreaker.java

```diff
@@ -48,6 +57,12 @@ public ChildMemoryCircuitBreaker(BreakerSettings settings, Logger logger, Hierar
         this.logger = logger;
         logger.trace(() -> format("creating ChildCircuitBreaker with settings %s", settings));
         this.parent = parent;
+        this.trippedCountMeter = telemetryProvider.getMeterRegistry()
+            .registerLongCounter(
```

Comment:
If this method is called twice, it will throw an exception.  That goes for anywhere `.register*` is used.  I'm not familiar with the context surrounding this class' lifecycle (incl. weird stuff like class loaders), but I've earned a bit of paranoia. 

In my PRs, I've started using a pattern whereby metrics are registered outside the class and passed in (https://github.com/elastic/elasticsearch/pull/101577/files#diff-74956e42dd1188b5474ec53450471777baa197a6bceb325716e31ec00d31c3f8R19-R48) .  Here's my reason for this pattern, athough you may arrive at a different conclusion: https://github.com/elastic/elasticsearch/pull/101577/files#r1379132909

Reply:
A bit more on what I'm getting at:  It's possible that even if we do everything correctly now, some poor soul may come along in the future and change the lifecycle guarantees of a class with `register*` in the constructor.  You can't write a test for this.  So to guard against this, I'd prefer to keep the logic that enforces correctness of metric registration separate, and thereby make pre-registered metrics a dependency for a class' construction.

Eventually we'll have metrics everywhere, so we might as well start off on the right footing; if not this, then something else to the same effect.

Reply:
I agree. Registration needs to be done once and if it is done in a constructor like here we have no guarantee it won't be used multiple times creating multiple objects trying to register the same metric.

Reply:
So I created a Metric class which is created when a node is constructed and that takes ownership of metrics registration. This ensures registration only happens once, when node construction happens.